### PR TITLE
[Database] Finalize Database

### PR DESCRIPTION
**Pull request by raineio**

### CODE REVIEW

**Review by wolfcomp — CHANGES_REQUESTED** [review]
Fix the different linting errors and it looks fine.

File: src/functions/random.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export function createRandomToken() {
+    let token = "";
+    const characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+    for (let i = 0; i < 10; i++) {
+        token += characters.charAt(Math.floor(Math.random() * characters.length));
```
Comment (wolfcomp) [inline src/functions/random.ts:5]: `Math.random` does not provide a [cryptographic stable](https://dev.to/darkmavis1980/random-numbers-in-node-js-with-crypto-4agg) token.

File: src/functions/random.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    }
+    return token;
+}
+
+export function createRandomState() {
+    let state = "";
+    const characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_.";
+    for (let i = 0; i < 30; i++) {
+        state += characters.charAt(Math.floor(Math.random() * characters.length));
```
Comment (wolfcomp) [inline src/functions/random.ts:14]: `Math.random` does not provide a [cryptographic stable](https://dev.to/darkmavis1980/random-numbers-in-node-js-with-crypto-4agg) state

**Review by wolfcomp — CHANGES_REQUESTED** [review]

File: src/api/login/login.ts [hunk 3]
```diff
@@ -160,7 +166,7 @@ export class BeatLeaderLogin {
                     code: code.toString(),
                     client_secret: process.env.BEATLEADER_SECRET,
                     client_id: process.env.BEATLEADER_ID,
-                    redirect_uri: `${process.env.REDIRECT_URI_API}/login/beatleader`,
+                    redirect_uri: "https://saberquest.xyz/login/beatleader",
```
Comment (wolfcomp) [inline src/api/login/login.ts:169]: Do not hard code redirect URIs

**Review by wolfcomp — CHANGES_REQUESTED** [review]

**Review by wolfcomp — APPROVED** [review]

**Review by StormPacer — APPROVED** [review]

**Pull request merged**